core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SubmissionApiServiceSpec.scala

```diff
@@ -386,22 +386,34 @@ class SubmissionApiServiceSpec extends ApiServiceSpec with TableDrivenPropertyCh
   //
   // and look for a section called "LAST DETECTED DEADLOCK".
   it should "not deadlock when aborting a large submission" in withLargeSubmissionApiServices { services =>
-    withWorkflowSubmissionActor(services) { _ =>
-      val wsName = largeSampleTestData.wsName
-      val mcName = MethodConfigurationName("no_input", "dsde", wsName)
-      val methodConf = MethodConfiguration(mcName.namespace, mcName.name, Some("Sample"), None, Map.empty, Map.empty, AgoraMethod("dsde", "no_input", 1))
-      val numIterations = 20
-
-      (1 to numIterations).map { i =>
-        logger.info(s"deadlock test: iteration $i/$numIterations")
-        val submission = createAndMonitorSubmission(wsName, methodConf, largeSampleTestData.sampleSet, Option("this.hasSamples"), services)
-
-        assertResult(numSamples) {
-          submission.workflows.size
-        }
+    withStatsD {
+      withWorkflowSubmissionActor(services) { _ =>
+        val wsName = largeSampleTestData.wsName
+        val mcName = MethodConfigurationName("no_input", "dsde", wsName)
+        val methodConf = MethodConfiguration(mcName.namespace, mcName.name, Some("Sample"), None, Map.empty, Map.empty, AgoraMethod("dsde", "no_input", 1))
+        val numIterations = 20
+
+        (1 to numIterations).map { i =>
+          logger.info(s"deadlock test: iteration $i/$numIterations")
+          val submission = createAndMonitorSubmission(wsName, methodConf, largeSampleTestData.sampleSet, Option("this.hasSamples"), services)
+
+          assertResult(numSamples) {
+            submission.workflows.size
+          }
 
-        abortSubmission(services, wsName, submission.submissionId, false)
+          abortSubmission(services, wsName, submission.submissionId, false)
+        }
+      }
+    } { capturedMetrics =>
+      var counter: Int = 0
```

Comment:
This seemed to be the only existing unit test that hit the bit of code I was touching. I'm not sure if it is OK to just piggyback on the existing test instead of writing a dedicated one (perhaps I should at least update the test description to include this metric testing), but to keep the test passing I at least need to include the "withStatsD" wrapper. And then it doesn't add any additional overhead to include the metric assert code here.